packages/twenty-front/src/modules/object-record/relation-picker/components/RelationPicker.tsx

```diff
@@ -39,11 +42,32 @@ export const RelationPicker = ({
     selectedEntity: EntityForSelect | null | undefined,
   ) => onSubmit(selectedEntity ?? null);
 
+  const { objectMetadataItem: relationObjectMetadataItem } =
+    useObjectMetadataItem({
+      objectNameSingular:
+        fieldDefinition.metadata.relationObjectMetadataNameSingular,
+    });
+
+  const relationFieldMetadataItem = relationObjectMetadataItem.fields.find(
+    ({ id }) => id === fieldDefinition.metadata.relationFieldMetadataId,
+  );
+
+  const { entityId } = useContext(FieldContext);
+
+  const { handleOnCreate } = useAddNewRecordAndOpenRightDrawer({
```

Comment:
handleOnCreate should be the responsibility of a component, useAddNewRecordAndOpenRightDrawer should just expose createNewRecordAndOpenRightDrawer.